### PR TITLE
Update budget and clean up code

### PR DESCRIPTION
**Pull request by ryan1604**
- Update singleton storage
- Add balance command

### CODE REVIEW

**Review by NeoMinWei — APPROVED** [review]
LGTM

**Review by wwweert123 — APPROVED** [review]
Thanks for cleaning up the code. Appreciate it

File: src/main/java/seedu/financialplanner/FinancialPlanner.java [hunk 1]
```diff
@@ -52,7 +49,7 @@ public void run() {
 
     public void save() {
         try {
-            storage.save(cashflowList, FILE_PATH);
+            storage.save(FILE_PATH);
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/FinancialPlanner.java:52]: Maybe file_path can be stored in the storage class?

File: src/main/java/seedu/financialplanner/commands/BudgetCommand.java [hunk 2]
```diff
@@ -53,38 +51,98 @@ public BudgetCommand(RawCommand rawCommand) throws FinancialPlannerException {
...
+        logger.log(Level.INFO, "Updating budget");
+        ui.printBudgetBeforeUpdate();
+        Budget.updateBudget(budget);
+        ui.printBudgetAfterUpdate();
+    }
+
+    private void setBudget() {
+        logger.log(Level.INFO, "Setting budget");
+        Budget.setBudget(budget);
+        ui.showMessage("A monthly budget of " + Budget.getInitialBudgetString()
```
Comment (wwweert123) [inline src/main/java/seedu/financialplanner/commands/BudgetCommand.java:145]: maybe extract this method further into showMonthlyBudget in UI?

**Review by wwweert123 — COMMENTED** [review]
Thanks for your hard work!

**Comment by NeoMinWei** [comment]
GOOD JOB

File: src/main/java/seedu/financialplanner/commands/BudgetCommand.java [hunk 3]
```diff
@@ -53,38 +51,98 @@ public BudgetCommand(RawCommand rawCommand) throws FinancialPlannerException {
...
-            ui.showMessage("A monthly budget of " + Budget.getInitialBudgetString()
-                    + " has been set.");
+            setBudget();
             break;
         case "update":
-            logger.log(Level.INFO, "Updating budget");
-            ui.printBudgetBeforeUpdate();
-            Budget.updateBudget(budget);
-            ui.printBudgetAfterUpdate();
+            updateBudget();
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/BudgetCommand.java:87]: maybe some logging can be added here

**Review by NeoMinWei — COMMENTED** [review]
LGTM

File: src/main/java/seedu/financialplanner/FinancialPlanner.java (hunk 1 above)
Comment (ryan1604) [reply src/main/java/seedu/financialplanner/FinancialPlanner.java:52]: i put file path in parameter for junit testing:D

File: src/main/java/seedu/financialplanner/commands/BudgetCommand.java (hunk 2 above)
Comment (ryan1604) [reply src/main/java/seedu/financialplanner/commands/BudgetCommand.java:145]: ok will note for the future

File: src/main/java/seedu/financialplanner/commands/BudgetCommand.java (hunk 3 above)
Comment (ryan1604) [reply src/main/java/seedu/financialplanner/commands/BudgetCommand.java:87]: it is in the method:)

**Pull request merged**